### DISCUSSION TITLE
My first experiences with SILE

### DISCUSSION THREAD

**Discussion by Brixy** [sile-typesetter/sile#1329] [General]
Having used (Lua)LaTeX for years, SILE seems extremely flexible and fast und getting started (writing a book) is a lot easier that with LaTeX.

The one thing I miss most is an article class (#455) with an example document showing all possible types of content.

The book class with `twoside`, `open-double-page` etc. is not a good fit for such use cases IMHO.

It would be great if such a class supported typical markdown inline and block elements, like (three levels?) of (unnumbered?) headings, footnotes, different types of lists … The planned Pandoc writer would be excellent.

I would really like to help here, but I’m afraid I can’t ;-)

So to just get started I came up with this primitive solution which is definitely not the SILE way. But the results look ok.

```
\begin{document}
\set[parameter=document.lskip,value=8%pw]
\set[parameter=document.rskip,value=8%pw]
\set[parameter=document.baselineskip,value=3ex]
\set[parameter=document.parskip,value=6pt plus 1pt]
\define[command=section]{\font[size=16pt,weight=600]{\bigskip\noindent\process}\medskip\set[parameter=current.parindent,value=0]}
\define[command=subsection]{\font[size=13pt,weight=600]{\bigskip\noindent\process}\medskip\set[parameter=current.parindent,value=0]}
…
```

Creating a new class with the bringhurst example leads to an error—even using the [extended example](https://github.com/sile-typesetter/sile/issues/1062#issuecomment-696565939).

```
attempt to index field 'frame' (a nil value)
```

Thanks for putting so much work and effort into SILE.

**Comment by Omikhleia** [comment]
Greetings @Brixy 
For the record, I could mention my [own repository](https://github.com/Omikhleia/omikhleia-sile-packages) where I have a wholly redesigned book class ("omibook") with plenty of nifty things (customizable sections, tables and figures, customizable table of contents, cross-references label and refs, ...); as well as an enumeration (list) package ("enumitem"). You might want to give it a try - Actually, an article class based on the same core principles is something I certainly want to add at some point.

**Comment by Brixy** [comment]
Thank you very much, @Omikhleia.

Your repo (and SILE’s capabilities) are tremendously impressive.

I will definitely try out your classes and packages.

Thanks a lot again!

**Comment by alerque** [comment]
Thanks for the feedback. Hearing initial impressions and the first places people hit friction is actually immensely useful as a developer. After using a system for some time it is easy to forget or loose sight of those things.

I still agree something in the "article" league may be useful and we'd be happy to see class to fit that space in the core SILE distribution. That being said I think one of the reasons it has languished is what goes in it is such a subjective call. Even it the TeX world there are quite a few choices in that space and quite a few different use cases for some of them. There is already some discussion of this on the related issue. If you'd like to help drive that forward probably the most useful thing to do is comment there with more feedback on exactly what should/should not be included. If we have a spec to work towards it shouldn't actually be that hard to implement.

The Pandoc conversion is still languishing in the "it works well enough for me" phase. My Pandoc fork has working sile writer support that covers the needs of most of my book projects, and my Haskell skills are not really good enough to easily get it polished up well enough to to upstream. I do need to give that another go... if you happen to know of anybody with Haskell skills we could convince to help that would be amazing.

**Comment by alerque** [comment]
I see we have an issue #1062 for the bringhurst example being broken.

As for further examples with more extensive usage, I think it might be best to open issues on the [website repository](https://github.com/sile-typesetter/sile-typesetter.github.io/issues/) for anything you'd like to see on the example front.
